modules/core/src/main/scala/ch/epfl/bluebrain/nexus/admin/core/resources/ResourceRejection.scala

```diff
@@ -7,14 +7,20 @@ import ch.epfl.bluebrain.nexus.commons.types.Rejection
   * Enumeration type for rejections returned when attempting to evaluate commands.
   */
 sealed trait ResourceRejection extends Rejection
+
 object ResourceRejection {
 
   /**
-    * Signals the failure to perform a resource modification due to payload shape constraint violations.
+    * Signals an error during the SHACL validation of a resource payload.
+    */
+  final case object ResourceValidationError extends ResourceRejection
```

Comment:
I would have something like
```scala
final case class InvalidResource(report: ValidationReport) extends ResourceRejection
```
and in some place you can have the encoding for it as:
```scala
  private[marshallers] implicit val validationReportEnc: Encoder[ValidationReport] =
    Encoder.instance(_.json)
```